opsimulator/opsimulator.go

```diff
@@ -143,12 +143,13 @@ func (opSim *OpSimulator) Stopped() bool {
 	return opSim.stopped.Load()
 }
 
-func (opSim *OpSimulator) startStartupTasks() {
+func (opSim *OpSimulator) addDependencies() error {
 	for _, chainID := range opSim.L2Config.DependencySet {
-		opSim.startupTasks.Go(func() error {
```

Comment:
if the dependency set is > 1, then this fails due to nonce issues. I did not see a way to set a nonce on a deposit tx, so I modified this to be a synchronous for loop.

Reply:
makes sense! we can fix later https://github.com/ethereum-optimism/supersim/issues/99